src/components/select-project/select-project.controller.ts

```diff
@@ -83,13 +83,18 @@ export class SelectProjectController implements angular.IController {
   private canIAddToProject(): boolean {
     let canIAddToProject: boolean = true;
 
-    // TODO AuthorizationService.canIAddToProject assumes that the ProjectsService.get has already completed performed
-    // but currently this happens after project change events fire and it happens in the various dialog controllers, not
-    // as part of this component.  For now always return true until we can resolve this.
-    //
-    // if (!this.isNewProject()) {
-    //   canIAddToProject = this.AuthorizationService.canIAddToProject(_.get(this.ctrl.selectedProject, 'metadata.name'));
-    // }
+    var projectName = _.get(this.ctrl.selectedProject, 'metadata.name');
+
+    if (!this.isNewProject()) {
+      if (_.isUndefined(this.AuthorizationService.getRulesForProject(projectName))) {
+        this.AuthorizationService.getProjectRules(projectName).then( () => {
+          canIAddToProject = this.AuthorizationService.canIAddToProject(projectName);
+          return this.ctrl.forms.selectProjectForm.selectProject.$setValidity('cannotAddToProject', canIAddToProject);
```

Comment:
I don't think the error gets cleared if I select "Create Project" after selecting a readonly project.

Reply:
It does get cleared, `canIAddToProject` is always initially set to true.

Reply:
Maybe I'm missing it, but I don't see where `$setValidity` is called again to clear the error even if `canIAddToProject` is set to true.

Reply:
The last line of the function, the return, will be called for a new project.

Reply:
Yup I see now, thanks